python/cudf/cudf/core/index.py

```diff
@@ -992,7 +994,7 @@ def as_index(arbitrary, **kwargs):
     kwargs = _setdefault_name(arbitrary, kwargs)
 
     if isinstance(arbitrary, Index):
-        return arbitrary.rename(**kwargs)
+        return arbitrary.rename(**kwargs, inplace=True)
```

Comment:
This will modify the name of the input index. You need to shallow copy arbitrary before doing the inplace rename.

Reply:
that's right - updating this.

Reply:
done